zrml/prediction-markets/src/lib.rs

```diff
@@ -335,7 +345,7 @@ mod pallet {
             ensure!(creator == sender, "Canceller must be market creator.");
             ensure!(status == MarketStatus::Proposed, "Market must be pending approval.");
             // The market is being cancelled, return the deposit.
-            CurrencyOf::<T>::unreserve(&creator, T::AdvisoryBond::get());
+            CurrencyOf::<T>::unreserve_named(&RESERVE_ID, &creator, T::AdvisoryBond::get());
```

Comment:
We need distinguished names/ids for the reservation, otherwise different reservations get mixed and slashed altogether Example: `AdvisoryBond` and `OracleBond` go in the same pot with id `RESERVE_ID`. If one user participates in both roles and fails one, both his reservations get slashed.
Suggestion: Per reservation type, use one unique id that ideally includes the pallet id if possible.

Reply:
I think this unified behavior is intended. For example, `create_(categorical|scalar)_market` sums up `AdvisoryBond` and `OracleBond`. @lsaether ?

Reply:
@sea212 https://github.com/zeitgeistpm/zeitgeist/issues/306 was opened to track this matter. In the meanwhile, I will merge this PR